File: solutions/java/log-levels/1/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        return logLine.split(": ")[1].strip();
    }

    public static String logLevel(String logLine) {
        var raw = logLine.split(": ")[0];
        return raw.substring(1, raw.length() - 1).toLowerCase();
    }

    public static String reformat(String logLine) {
        return String.format("%s (%s)", message(logLine), logLevel(logLine));
    }
}

File: solutions/fsharp/log-levels/1/LogLevels.fs
module LogLevels

let message (logLine: string): string =
    logLine.Split(':').[1].Trim() in
    

let logLevel(logLine: string): string = 
    let s = logLine.Split(':').[0].Trim() in
    let n = s.Length in
    s[1..n-2].ToLower()
let reformat(logLine: string): string = 
    let a = message logLine in
    let b = logLevel logLine in
    sprintf "%s (%s)" a b